sys/arm64/include/frame.h

```diff
@@ -42,12 +42,24 @@
  * NOTE: keep this structure in sync with struct reg and struct mcontext.
  */
 struct trapframe {
+#if __has_feature(capabilities)
+	uintcap_t tf_sp;
```

Comment:
I believe we now define uintcap_t unconditionally (to uint64_t if it isn't supported), so we could drop the else and only ifdef the addition of `tf_ddc`.

Reply:
Unless this is yet another case of unnecessary stdint.h/stddef.h duplication for each architecture... In which case we should just add uintcap_t to an sys/_types.h

Reply:
Where is that typedef? The riscv trapframe defines this in a similar way.

Reply:
sys/sys/_types.h
```c
#if !__has_feature(capabilities)
typedef	__intptr_t	__intcap_t;
typedef	__uintptr_t	__uintcap_t;
#endif
```

sys/sys/_stdint.h (added in 2018, 8a64dd37f2ae722a41fca70383d53c2b0b55c70a)
```c
#ifndef _INTCAP_T_DECLARED
typedef	__intcap_t		intcap_t;
#define	_INTCAP_T_DECLARED
#endif
#ifndef _UINTCAP_T_DECLARED
typedef	__uintcap_t		uintcap_t;
#define	_UINTCAP_T_DECLARED
#endif
```

And that file also defines kintcap_t/kuintcap_t for things that should only be __(u)intcap_t if _KERNEL is defined.

Reply:
Can I wait until after this is merged for this given it's similar to riscv?

Reply:
Reducing ifdefs is always nice, but I don't really mind either way.